src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -26,5 +68,61 @@ private static bool GetSwitchValue(string switchName, string envVariable)
 
             return ret;
         }
+
+        private static bool TryGetStringValue(string switchName, string envVariable, [NotNullWhen(true)] out string? value)
+        {
+            value = AppContext.GetData(switchName) as string;
+            if (string.IsNullOrEmpty(value))
+            {
+                value = Environment.GetEnvironmentVariable(envVariable);
+                if (string.IsNullOrEmpty(value))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static string CreateLibraryName(ReadOnlySpan<char> baseName, ReadOnlySpan<char> suffix, ReadOnlySpan<char> extension, ReadOnlySpan<char> version, bool versionAtEnd = false)
+        {
+            int length = baseName.Length + suffix.Length + version.Length + extension.Length;
+
+            // We validate that suffix and version are not larger than 53 characters.
+            Span<char> result = stackalloc char[length];
+            baseName.CopyTo(result);
+
+            Span<char> secondPart = result.Slice(baseName.Length);
+            suffix.CopyTo(secondPart);
+
+            Span<char> middle = secondPart.Slice(suffix.Length);
+
+            if (!versionAtEnd)
+            {
+                version.CopyTo(middle);
+
+                Span<char> end = middle.Slice(version.Length);
+                extension.CopyTo(end);
+            }
+            else
+            {
+                extension.CopyTo(middle);
+
+                Span<char> end = middle.Slice(extension.Length);
+                version.CopyTo(end);
+            }
+
+            return result.ToString();
+        }
+
+        private static IntPtr LoadLibrary(string library, Assembly assembly, bool failOnLoadFailure)
+        {
+            if (!NativeLibrary.TryLoad(library, assembly, DllImportSearchPath.ApplicationDirectory, out IntPtr lib) && failOnLoadFailure)
+            {
+                Environment.FailFast($"Failed to load app-local ICU: {library}");
+            }
+
+            return lib;
```

Comment:
>lib [](start = 19, length = 3)

does NativeLibrary.TryLoad initialize lib IntPtr.Zero when it fail? just checking.

Reply:
Yeah, actually its return bool is the result of `lib != IntPtr.Zero`.

https://github.com/dotnet/runtime/blob/89d08de5d2e29c45a904513dcc554aa31f3a5d2e/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs#L122-L136